crowbar_framework/app/models/node_object.rb

```diff
@@ -1585,17 +1585,24 @@ def unique_device_for(device)
       end
       result = disk_lookups.map do |type|
         if meta["disks"][type] and not meta["disks"][type].empty?
-          "#{type}/#{meta["disks"][type].first}"
+          meta["disks"][type]
         end
       end
-
-      # virtio disk might have neither by-path nor by-id links, use the /dev/vdX
-      # name in that case
-      if result.empty?
-        "/dev/#{device}"
-      else
-        "/dev/disk/#{result.compact.first}"
-      end
+      unless result.empty?
+        match = result[0].find { |b| b =~ /^wwn-/ } ||
+          result[0].find { |b| b =~ /^scsi-[a-zA-Z]/ } ||
+          result[0].find { |b| b =~ /^scsi-[^1]/ } ||
+          result[0].find { |b| b =~ /^scsi-/ } ||
+          result[0].find { |b| b =~ /^ata-/ } ||
+          result[0].find { |b| b =~ /^cciss-/ } ||
+          result[0].first
+        # virtio disk might have neither by-path nor by-id links, use the /dev/vdX
+        # name in that case
+        if match.empty?
+          "/dev/#{device}"
+        else
+          "/dev/disk/#{type}/#{match}"
+        end
     else
```

Comment:
So you restorered that `else`, but there's no `else` for ` unless result.empty?` anymore? Is that intended?